### PR TITLE
Current Product Widget

### LINKED ISSUE #1454: Current Product Widget
**Is your feature request related to a problem? Please describe.**
The widget that indicates the user's current plan with links to the subscription management modal needs to be created.

**Describe the solution you'd like**
The component should be pretty simple, just some buttons for navigation within that settings page, storage usage components exists in the codebase already. Clicking change plan should trigger the plan selection modal in #1455 

When the user is on a Basic plan, the `Cancel` button should be disabled.

**Additional context**
image

### PR DESCRIPTION
**Pull request by RyRy79261**
close #1454

### CODE REVIEW

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx [hunk 1]
```diff
@@ -1,140 +1,121 @@
...
             >
-              <Button
-                variant="primary"
-                className={classes.changePlanButton}
-              >
-                <Trans>Buy more storage</Trans>
-              </Button>
-            </Link>
-          </div>
+            Change Plan
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx]:
```suggestion
            <Trans>Change Plan</Trans>
```

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx [hunk 2]
```diff
@@ -1,140 +1,121 @@
...
+    </Typography>
+    {
+      currentSubscription
+        ?  <Typography
+          variant="h5"
+          component="h5"
+        >
+          {
+            currentSubscription?.product.name
+          }
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx]:
```suggestion
          {currentSubscription?.product.name}
```

File: packages/files-ui/src/Components/Modules/Settings/index.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React, { useCallback, useMemo } from "react"
 import ProfileTab from "./ProfileTab"
-import { Tabs as TabsOrigin,
+import { Tabs,
   TabPane as TabPaneOrigin,
   Typography, Divider,
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/index.tsx]:
```suggestion
import {
  Tabs,
  TabPane as TabPaneOrigin,
  Typography,
  Divider,
```

File: packages/files-ui/src/Components/Modules/Settings/index.tsx [hunk 4]
```diff
@@ -136,10 +128,9 @@ const useStyles = makeStyles(({ constants, breakpoints, palette }: ITheme) =>
 
 const Settings: React.FC = () => {
   const { desktop } = useThemeSwitcher()
-  const { path = desktop ? "profile" : undefined } = useParams<{path: SettingsPath}>()
+  const { path = desktop ? "profile" : "" } = useParams<{path: SettingsPath}>()
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/index.tsx]: This breaks the typing and obliged you to add a hack below. Any reason to change this?
```suggestion
  const { path = desktop ? "profile" : undefined } = useParams<{path: SettingsPath}>()
```

File: packages/files-ui/src/Components/Modules/Settings/index.tsx [hunk 5]
```diff
@@ -171,7 +163,7 @@ const Settings: React.FC = () => {
         <div className={classes.tabsContainer}>
           <Tabs
             activeKey={path}
-            onTabSelect={onSelectTab}
+            onTabSelect={onSelectTab as (key: string) => void}
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/index.tsx]: The typing was all right afaik
```suggestion
            onTabSelect={onSelectTab}
```

**Review by Tbaut — COMMENTED** [review]
A couple nits, looks good otherwise from what I could test

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx [hunk 6]
```diff
@@ -1,140 +1,121 @@
...
+        : <Loading />
+    }
+    {storageSummary &&
+      <>
+        <div className={classes.spaceUsedBox}>
+          <Typography
+            variant="body2"
+            component="p"
+          >
+            {`${formatBytes(storageSummary.used_storage, 2)} of ${formatBytes(
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx]: needs translation, not sure how, with `t` should work
```suggestion
            {t`${formatBytes(storageSummary.used_storage, 2)} of ${formatBytes(
```

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx [hunk 7]
```diff
@@ -1,140 +1,121 @@
...
     }
   })
 )
 
-const CurrentProduct: React.FC = () => {
+interface ISubscriptionWidget {
+  className?: string
+}
+
+const CurrentProduct = ({ className }: ISubscriptionWidget) => {
```
Comment (FSM1) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/CurrentProduct.tsx]:
```suggestion
interface ICurrentProduct {
  className?: string
}

const CurrentProduct = ({ className }: ICurrentProduct ) => {
```

**Review by FSM1 — APPROVED** [review]
Great stuff 💯

File: packages/files-ui/src/Components/Modules/Settings/index.tsx (hunk 4 above)
Comment (RyRy79261) [reply packages/files-ui/src/Components/Modules/Settings/index.tsx]: Wasn't a deliberate change, the merges made it difficult to figure out what was a required change or not

File: packages/files-ui/src/Components/Modules/Settings/index.tsx (hunk 5 above)
Comment (RyRy79261) [reply packages/files-ui/src/Components/Modules/Settings/index.tsx]: Was throwing for me

Comment (Tbaut) [reply packages/files-ui/src/Components/Modules/Settings/index.tsx]: It should not anymore now that you applied my previous changes, with the undefined changed to empty string :)

**Review by Tbaut — APPROVED** [review]

**Pull request merged**